config/docker/rustc-1.62.jinja2

```diff
@@ -0,0 +1,42 @@
+{% extends 'clang-15.jinja2' %}
+
+{% block packages %}
+{{ super() }}
+
+ENV RUST_VER=1.62.0
+ENV BINDGEN_VER=0.56.0
+
+ENV RUST_TRIPLE=rust-${RUST_VER}-x86_64-unknown-linux-gnu
+
+ENV CARGO_HOME=/usr/$RUST_TRIPLE/cargo
+
+ENV PATH=/usr/$RUST_TRIPLE/bin:${PATH}
+
+# fetch, validate and import the rust team key
+RUN wget https://static.rust-lang.org/rust-key.gpg.ascii
+RUN gpg --import --import-options show-only rust-key.gpg.ascii | \
+        grep -q 108F66205EAEB0AAA8DD5E1C85AB96E6FA1BE5FE
+RUN gpg --import rust-key.gpg.ascii
+
+# fetch, verify the toolchain
+RUN wget https://static.rust-lang.org/dist/${RUST_TRIPLE}.tar.gz
+RUN wget https://static.rust-lang.org/dist/${RUST_TRIPLE}.tar.gz.asc
+RUN gpg --verify ${RUST_TRIPLE}.tar.gz.asc ${RUST_TRIPLE}.tar.gz
+
+# install & cleanup tmp files
+RUN tar -xf ${RUST_TRIPLE}.tar.gz  -C /tmp/
+RUN /tmp/${RUST_TRIPLE}/install.sh --prefix=/usr/${RUST_TRIPLE} \
+        --components=rustc,cargo,rust-std-x86_64-unknown-linux-gnu
+RUN rm -rf /tmp/${RUST_TRIPLE}
+RUN rm /${RUST_TRIPLE}*
+
+RUN mkdir $CARGO_HOME
+RUN printf '[build]\nrustflags = ["-C", "linker=clang"]' > $CARGO_HOME/config.toml
```

Comment:
Why is this needed? A comment would be useful here.

Reply:
> but that cc is being used as the linker

Yeah, that is what I meant by `cc` instead of `ld`.

> I tried setting for eg linker-flavor=ld.lld but got the following error because libgcc is still required.

As far as I know, that is because `linker-flavor` will make it call `lld` directly, while C compilers take care of adding some extra flags to the linker, such as to find the system libraries.